ouroboros-consensus/src/Ouroboros/Consensus/Node/Run.hs

```diff
@@ -99,3 +100,6 @@ class ( LedgerSupportsProtocol           blk
       , ShowProxy                 (Query blk)
       , ShowProxy           (TxId (GenTx blk))
       ) => RunNode blk
+  -- This class is intentionally empty. It is not necessarily compositional, but
+  -- we want as many of its super classes as possible to rely on compositional
+  -- instances when possible. Not putting any methods here helps encourage that.
```

Comment:
I don't quite get what the comment is saying here. What do we understand by "the typeclass is not necessarily compositional" and "we want as many of its super classes as possible to rely on compositional"?

Reply:
Yeah, "compositionality" is pretty high-level/vague. So I added mention of a slightly concrete example to the comment: the HFC `HardForkBlock`. Thanks.